src/coreclr/jit/importercalls.cpp

```diff
@@ -2117,7 +2117,20 @@ void Compiler::impPopArgsForSwiftCall(GenTreeCall* call, CORINFO_SIG_INFO* sig,
 
     if (swiftErrorIndex != sig->numArgs)
     {
-        *swiftErrorArg = call->gtArgs.GetArgByIndex(swiftErrorIndex);
+        // Before calling a Swift method that may throw, the error register must be cleared,
+        // as we will check for a nonzero error value after the call returns.
+        // By adding a well-known "sentinel" argument that uses the error register,
+        // the JIT will emit code for clearing the error register before the call,
+        // and will mark the error register as busy so it isn't used to hold the function call's address.
+        CallArg* const swiftErrorArg          = call->gtArgs.GetArgByIndex(swiftErrorIndex);
+        GenTree* const errorSentinelValueNode = gtNewIconNode(0);
+        call->gtArgs.InsertAfter(this, swiftErrorArg,
+                                 NewCallArg::Primitive(errorSentinelValueNode).WellKnown(WellKnownArg::SwiftError));
+
+        // Swift call isn't going to use the SwiftError* arg, so don't bother emitting it
+        assert(swiftErrorNode != nullptr);
+        *swiftErrorNode = swiftErrorArg->GetNode();
+        call->gtArgs.Remove(swiftErrorArg);
```

Comment:
Do we need to potentially adjust `swiftSelfIndex` here? Or perhaps instead delay the removal until after the loop below?

Reply:
Good catch: If the `SwiftSelf` arg comes after the `SwiftError*` arg, then yes, we do need to decrement it. It seems easier to delay the removal. Fixed.